include/library.h

```diff
@@ -439,6 +439,7 @@ class Library: public std::enable_shared_from_this<Library>
   AttributeCounts getBookAttributeCounts(BookStrPropMemFn p) const;
   std::vector<std::string> getBookPropValueSet(BookStrPropMemFn p) const;
   BookIdCollection filterViaBookDB(const Filter& filter) const;
+  std::string getBestTargetBookId(const Bookmark& bookmark) const;
```

Comment:
During rebase&history-cleanup - it's better if this commit is squashed too (so that the implementation of `migrateBookmarks()` is immediately introduced in the most comprehensible way).